paddle/phi/core/distributed/comm_context_manager.cc

```diff
@@ -67,7 +77,29 @@ void CommContextManager::CreateNCCLCommContext(
 
   auto nccl_comm_context =
       std::make_unique<NCCLCommContext>(rank, size, nccl_id);
-  auto& comm_context_manager = CommContextManager::GetInstance();
+
+  if (CommContextManager::device_id != -1) {
```

Comment:
why not use dev_ctx from global dev_ctx pool?

Reply:
`NCCLCommContext` holds `std::unique_ptr<phi::GPUContext> dev_ctx`, so we create a new object out of `DeviceContextPool` control.